runtime/src/main/java/com/flipkart/flux/deploymentunit/DeploymentUnitUtil.java

```diff
@@ -0,0 +1,212 @@
+        List<String> classNames = (List<String>) yamlConfiguration.getProperty(WORKFLOW_CLASSES);
+        Set<Method> methods = new HashSet<>();
+
+        //loading this class separately in this class loader as the following isAnnotationPresent check returns false, if
+        //we use default class loader's Task, as both class loaders don't have any relation between them.
+        Class taskClass = urlClassLoader.loadClass(Task.class.getCanonicalName());
+
+        for(String name : classNames) {
+            Class clazz = urlClassLoader.loadClass(name);
+
+            for(Method method : clazz.getMethods()) {
+                if(method.isAnnotationPresent(taskClass))
+                    methods.add(method);
+            }
+        }
+
+        return methods;
+    }
+
+    /**
+     * Given a deployment unit deployment unit name retrieves properties from config file and returns them.
+     * @param deploymentUnitName
+     * @return
+     * @throws IOException
+     */
+    public YamlConfiguration getProperties(String deploymentUnitName) throws IOException {
+        URLClassLoader classLoader = getClassLoader(deploymentUnitName);
+        return getProperties(classLoader);
+    }
+
+    /**
+     * Given a deployment unit classloader retrieves properties from config file and returns them.
+     * @param urlClassLoader
+     * @return YamlConfiguration
+     * @throws IOException
+     */
+    public YamlConfiguration getProperties(URLClassLoader urlClassLoader) throws IOException {
+        return new YamlConfiguration(urlClassLoader.getResource(CONFIG_FILE));
+    }
+
+    /**
+     * Provides {@link java.net.URLClassLoader for a given deployment unit}
+     */
+    static class ClassLoaderProvider {
+
+        /**
+         * Builds and returns URLClassLoader for a provided deployment unit.
+         * This assumes the provided deployment unit directory has the following structure.
+         *
+         * DeploymentUnit
+         * |_____ main              //contains main jar
+         * |_____ lib               //contains libraries on which main jar is depending
+         * |_____ flux_config.yml   //properties file
+         *
+         * The constructed class loader is independent and doesn't share anything with System class loader.
+         * @param deploymentUnitFQN
+         * @return URLClassLoader
+         * @throws java.net.MalformedURLException
+         */
+        public static URLClassLoader getClassLoader(String deploymentUnitFQN) throws MalformedURLException, FileNotFoundException {
+
+            File[] mainJars = new File(deploymentUnitFQN+"main").listFiles();
+            File[] libJars = new File(deploymentUnitFQN+"lib").listFiles();
+
+            if(mainJars == null)
+                throw new FileNotFoundException("Unable to build class loader. Required directory "+deploymentUnitFQN+"main is empty.");
+
+            File configFile = new File(deploymentUnitFQN+CONFIG_FILE);
+            if(!configFile.isFile())
```

Comment:
1. done.
2. Both serves the same purpose, if the file doesn't exist this if would be executed. I think it is good to use isFile() and throw error before itself instead of throwing at file parsing time.